dp_mobility_report/model/preprocessing.py

```diff
@@ -116,19 +138,37 @@ def assign_points_to_tessellation(
     )
 
     # Spatial join points to polygons
-    df = gpd.sjoin(
+    df = gpd.sjoin_nearest(
         tessellation[[const.TILE_ID, const.TILE_NAME, const.GEOMETRY]],
         gdf,
-        how="right",
+        how="right"
     )
     df.drop(["index_left", const.GEOMETRY], axis=1, inplace=True)
     return pd.DataFrame(df)
 
 
 def sample_trips(
-    df: pd.DataFrame, max_trips_per_user: int, user_privacy: bool
+    df: pd.DataFrame, max_trips_per_user: int, user_privacy: bool, timestamps: bool
 ) -> pd.DataFrame:
-    if user_privacy:
+    if user_privacy and timestamps:
```

Comment:
I dont see a difference between the two conditions? The sampling should not rely on the timestamp